evm/db/state.py

```diff
@@ -294,7 +316,13 @@ def account_is_empty(self, address):
     # Internal
     #
     def _get_account(self, address):
-        rlp_account = self._trie[address]
+        cache_key = (self._unwrapped_db, self.root_hash, address)
+        if cache_key in account_cache:
+            rlp_account = account_cache[cache_key]
+        else:
+            rlp_account = self._trie[address]
+            account_cache[cache_key] = rlp_account
```

Comment:
I've been a fan of the following pattern for cached functions.  I find it easier to parse.

```python
def cached_fn():
    if key not in cache:
        cache[key] = get_value()

    value = cache[key]
```

It's a minor improvement but it's a smaller conditional block to parse and the *value* always comes from the same place.

Reply:
I like that!